test/brontide-test.js

```diff
@@ -13,6 +13,159 @@ const PROLOGUE = 'lightning';
  * Tests
  */
 
+describe('CipherState', function() {
+  const cipher = new CipherState();
+  const key = Buffer.from(
+    '2121212121212121212121212121212121212121212121212121212121212121',
+    'hex'
+  );
+  const salt = Buffer.from(
+    '1111111111111111111111111111111111111111111111111111111111111111',
+    'hex'
+  );
+
+  it('should initalize with the key and salt', () => {
+    cipher.initSalt(key, salt);
+
+    assert.bufferEqual(cipher.key, key);
+
+    assert.bufferEqual(cipher.salt, salt);
+  });
+
+  it('should rotate the secret key', () => {
+    cipher.rotateKey();
+
+    assert.bufferEqual(
+      cipher.key,
+      Buffer.from(
+        '0b579ba44366e4d49ac7a44a8203925cb6d610e950aee7a23c47a5448173af11',
+        'hex'
+      )
+    );
+
+    assert.bufferEqual(
+      cipher.salt,
+      Buffer.from(
+        'be23775b41e7c67d1ec6dcfc21299f32461e145d4164f65943b4b99fcaff6dee',
+        'hex'
+      )
+    );
+
+    assert.strictEqual(cipher.nonce, 0);
+  });
+
+  it('should properly encrypt given text and empty ad', () => {
+    // Reset the cipher
+    const cipher = new CipherState();
+    cipher.initSalt(key, salt);
+
+    const hello = Buffer.from('hello', 'ascii');
+
+    const tag = cipher.encrypt(hello);
+
+    assert.bufferEqual(
+      tag,
+      Buffer.from('f11ae60b9df4c6ea25aea58ce1b6df83', 'hex')
+    );
+    assert.bufferEqual(hello, Buffer.from('0935b4c530', 'hex'));
+
+    // Round 2
+    const hello2 = Buffer.from('hello', 'ascii');
```

Comment:
Could we use `HELLO`, like above comment?

Reply:
See Above